server/src/main/java/org/apache/iotdb/db/utils/datastructure/BinaryTVList.java

```diff
@@ -58,6 +62,12 @@ public void putBinary(long timestamp, Binary value) {
     if (sorted && rowCount > 1 && timestamp < getTime(rowCount - 2)) {
       sorted = false;
     }
+    rawSize += getBinarySize(value);
```

Comment:
This should already be estimated before write, store the memory in Binary and use here